src/InputParser.jl

```diff
@@ -0,0 +1,374 @@
+module InputParser
+
+# Julia implicitly numbers Enum items 0,1,2,3,4 (in order)
+# I put them in same order as FORTRAN code (i.e. NOPT=0 meant
+# ConsolidationSwell so Int(ConsolidationSwell) = 0) however
+# this wont matter since we should never be referring to the integer 
+# values explicitly anyways
+@enum Model ConsolidationSwell LeonardFrost Schmertmann CollapsibleSoil SchmertmannElastic 
+
+# These are variables/objects visible to all modules
+# that include InputParser module
+export Model, InputData
+
+# This code repeats many times in InputData constructor
+# Parses String at input[index] and returns an array A such that size(A) == items
+function parseCurrentLine(input::Array{String}, items::Int, index::Int)
+    currentLine = input[index]
+    # Tabs ruin splitting by space
+    currentLine = replace(currentLine, "\t" => " ")
+    currentLineData = split(currentLine, " ")
+    currentLineData = filter(x -> x != "", currentLineData)
+    # Check if data is there
+    if size(currentLineData)[1] < items
+        println("Error: Invalid input file!")
+        println("\t>Line $(index): $(items) values expected")
+        return -1
+    end
+    return currentLineData
+end
+
+# TODO: Search for error handling (throws Exception/ try-catch block)
```

Comment:
Yes, we want to be as robust as possible.  We should eventually add test cases that verify that the error handling works.  We will also be adding constraints on the inputs, since some of them will have a range of physically meaningful values.  (For instance Poisson's ration has to lie between 0.5 and 1.0.  Anything outside that range is an error.  We should check the validity of the values on input (and we should have test cases to check we implemented everything correctly.)